lib/filesystem.nix

```diff
@@ -42,4 +44,13 @@
               type = (builtins.readDir parent).${base} or null;
           in file == /. || type == "directory";
     in go (if isDir then file else parent);
+
+  mapDir = filter: f: path:
```

Comment:
Should be called something like `filterMapDir`

Reply:
I think I saw something for recursing and filtering a directory tree somewhere, I should probably look at that, if anyone remembers what it was.

Reply:
Maybe it *was* just `locateDominatingFile` from this same file. I suppose it would make sense to factor out the `mapDir` equivalent component. I didn't even notice the similarities when I was making the PR.

Reply:
Ok I added some documentation. Is it good?